src/Microsoft.Framework.ConfigurationModel/ConfigurationPathComparer.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Framework.ConfigurationModel
+{
+    public class ConfigurationPathComparer : IComparer<string>
```

Comment:
I think `path` is a concept used by JSON.NET and we don't have such a concept in our configuration system. `settings:address:ip:0` is a key in our understanding.